sound/soc/soc-pcm.c

```diff
@@ -1264,6 +1264,9 @@ static void dpcm_be_reparent(struct snd_soc_pcm_runtime *fe,
 
 		fe_substream = snd_soc_dpcm_get_substream(dpcm->fe, stream);
 		be_substream->runtime = fe_substream->runtime;
+		/* Make sure private_data from the old FE is not lost. */
+		fe_substream = snd_soc_dpcm_get_substream(fe, stream);
+		be_substream->runtime->private_data = fe_substream->runtime->private_data;
```

Comment:
FWIW. I looked at that and I removed the use of the private data in the BE in https://github.com/thesofproject/linux/pull/3908

I think it's completely wrong to share data from the FE in the BE. It should not be allowed, that's a design conceptual bug in DPCM.

In other words, the SOF parts don't care about this any longer.

Reply:
I do agree. If there is no explicitly defined place where data between FE and BE can be exchanged, doing so implicitly through private_data sharing is not the correct way to handle things.

This problem, as most, is probably best addressed incrementally - fix the bug, then work on removing the flaw entirely.